File: DatalakeClient/src/@types/enums/LogType.ts
export enum LogType {
	Trace = 0,
	Information = 1,
	Success = 2,
	Warning = 3,
	Error = 4,
}

export function LogTypeDescription(type: LogType) {
	switch (type) {
		case LogType.Trace: return 'отладка'
		case LogType.Information: return 'инфо'
		case LogType.Success: return 'успех'
		case LogType.Warning: return 'предупреждение'
		case LogType.Error: return 'ошибка'
		default: return '?'
	}
}

File: DatalakeApiClasses/Enums/LogType.cs
using System.Text.Json.Serialization;

namespace DatalakeApiClasses.Enums;

/// <summary>
/// Степень важности сообщения
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum LogType
{
	/// <summary>
	/// Отладка
	/// </summary>
	Trace = 0,

	/// <summary>
	/// Информирующее
	/// </summary>
	Information = 1,

	/// <summary>
	/// Сигнализация о успешном выполнении операции
	/// </summary>
	Success = 2,

	/// <summary>
	/// Предупреждение
	/// </summary>
	Warning = 3,

	/// <summary>
	/// Ошибка
	/// </summary>
	Error = 4,
}
